### PR TITLE
Changes for the next release, detailed list of changes:

### PR DESCRIPTION
**Pull request by jamesmunns**
* Update for 2018 edition
* Harmonize versions to 0.6.0-beta.1, matching -pac crates
* Update dependencies
* Massage some differences between 832 and 840 to be more consistent
* Remove Xargo.toml from 840
* Update my email address

All PRs in this family:

* [nrf52832-pac]
* [nrf52840-pac]
* [nrf52-hal]

[nrf52832-pac]: https://github.com/nrf-rs/nrf52832-pac/pull/11
[nrf52840-pac]: https://github.com/nrf-rs/nrf52840-pac/pull/2
[nrf52-hal]: https://github.com/nrf-rs/nrf52-hal/pull/32

### CODE REVIEW

**Comment by jamesmunns** [comment]
Hey @nrf-rs/nrf52, I believe this is ready to land now. I have already cut a release and published on crates.io as `0.6.0-beta.1`. Could you please test with your crates locally, and let me know how it goes?

I'd like to merge this sooner than later, and if we need to iterate on `master` before we cut the final `0.6.0` release, that works for me.

Please see the related issues linked above for this family of PRs.

**Comment by jamesmunns** [comment]
Fixing adafruit crate now...

**Comment by jamesmunns** [comment]
Note: published `0.6.0-beta.2` with @hannobraun's Timer additions, yanked `0.6.0-beta.1` of the HAL crates.

**Comment by jamesmunns** [comment]
Whoops, totally forgot these changes @hannobraun. You're right, that was an oversight. I will fix that shortly.

**Comment by jamesmunns** [comment]
This is now fixed. Once the `-pac` crates land (and versions are updated), this should be ready to merge.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: rust
 rust:
-  - nightly
```
Comment (hannobraun) [inline .travis.yml]: Any reason for not building on all channels?

Comment (jamesmunns) [reply .travis.yml]: No good ones! I'll add that back.

File: nrf52-hal-common/Cargo.toml [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
 
-[dependencies.embedded-hal]
-features = ["unproven"]
-version = "0.2.1"
-
-[dependencies.nrf52]
+[dependencies.nrf52832-pac]
 optional = true
-version = "~0.4"
+version = "=0.6.0"
```
Comment (hannobraun) [inline nrf52-hal-common/Cargo.toml]: Why the strict requirement? Seems like that will cause trouble down the line sooner or later (when we release a new `nrf52832-pac` and forget to update the dependency here).

File: nrf52-hal-common/Cargo.toml [hunk 3]
```diff
@@ -1,41 +1,48 @@
...
 
-[dependencies.nrf52840]
+[dependencies.nrf52840-pac]
 optional = true
-version = "~0.2"
+version = "=0.6.0"
```
Comment (hannobraun) [inline nrf52-hal-common/Cargo.toml]: Same comment about strict requirement applies here, too.

File: nrf52832-hal/Cargo.toml [hunk 4]
```diff
@@ -1,38 +1,44 @@
...
 
 [dependencies.cast]
 default-features = false
 version = "0.2.2"
 
+[dependencies.nrf52-hal-common]
+path = "../nrf52-hal-common"
+default-features = false
+features = ["52832"]
+version = "=0.6.0"
```
Comment (hannobraun) [inline nrf52832-hal/Cargo.toml]: Same comment about strict requirement applies here, too.

File: nrf52840-hal/Cargo.toml [hunk 5]
```diff
@@ -1,39 +1,46 @@
...
+edition = "2018"
 
 [dependencies]
-bare-metal = "0.2.0"
-cortex-m = "0.5.2"
-cortex-m-rt = "0.5.1"
+cortex-m = "0.5.8"
 nb = "0.1.1"
-nrf52840 = "0.2.0"
+nrf52840-pac = "=0.6.0"
```
Comment (hannobraun) [inline nrf52840-hal/Cargo.toml]: Same comment about strict requirement applies here, too.

File: nrf52840-hal/Cargo.toml [hunk 6]
```diff
@@ -1,39 +1,46 @@
...
 
 [dependencies.cast]
 default-features = false
 version = "0.2.2"
 
 [dependencies.nrf52-hal-common]
 path = "../nrf52-hal-common"
+default-features = false
 features = ["52840"]
+version = "=0.6.0"
```
Comment (hannobraun) [inline nrf52840-hal/Cargo.toml]: Same comment about strict requirement applies here, too.

File: nrf52-hal-common/Cargo.toml (hunk 2 above)
Comment (jamesmunns) [reply nrf52-hal-common/Cargo.toml]: Good call! Removed (and all following)

**Comment by jamesmunns** [comment]
@hannobraun I appreciate the last minute review, feel free to yell stop whenever :)

**Comment by jamesmunns** [comment]
@hannobraun any last thoughts?

**Comment by hannobraun** [comment]
@jamesmunns Sorry, I stopped seeing the notifications for some reason, and when I noticed the backlog, you were done and I was ready for bed. I didn't have time for a full review, but I didn't see anything objectionable.

Thank you, and good job!

**Pull request merged**